app/controllers/jobseekers/job_applications/build_controller.rb

```diff
@@ -38,20 +39,27 @@ def back_path
   end
 
   def form
-    @form ||= form_class.new(form_attributes)
+    @form ||= begin
+      attributes = form_attributes
+      attributes.merge!(jobseeker_profile_attributes) if step == :professional_status
```

Comment:
Suggestion: 

If I'm reading this right. The `form_attributes` is already merging a set of params only if the step is professional_status  [here](https://github.com/DFE-Digital/teaching-vacancies/pull/7056/files#diff-6fe827802e214e8b154f45038828bf184bafafb495542a1b0bb111b48659b01fR61):
```ruby
attributes.merge!(trn_params) if step == :professional_status
```

This line also conditionally merges the jobseeker profile attributes if the step is professional status.

I personally would find this logic easier to read/follow if both `merge!` calls were together under a single `if step == :professional_status` condition.

I'm leaving it as a suggestion you may want to explore. But as long as it works... this is just something I noticed that maybe would be simpler :shrug: